zcash_primitives/src/note_encryption.rs

```diff
@@ -80,6 +81,27 @@ impl PartialEq for Memo {
     }
 }
 
+impl Eq for Memo {}
+
+impl PartialOrd for Memo {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        let self_utf8 = self.to_utf8();
+        let other_utf8 = other.to_utf8();
+        match (self_utf8, other_utf8) {
+            (Some(Ok(a)), Some(Ok(b))) => a.partial_cmp(&b),
+            (Some(Ok(_)), _) => Some(Ordering::Less),
+            (_, Some(Ok(_))) => Some(Ordering::Greater),
+            (_, _) => self.0[..].partial_cmp(&other.0[..]),
+        }
```

Comment:
Memos are allowed to be non-UTF-8 if they start with a byte >= 0xF5. In any case the validity is not enforced by consensus.

Reply:
Yeah, this is also related to the source of the roundtrip errors; this is going to be reverted.